lib/spack/llnl/util/filesystem.py

```diff
@@ -893,6 +908,27 @@ def mkdirp(*paths, **kwargs):
             raise OSError(errno.EEXIST, "File already exists", path)
 
 
+def longest_existing_parent(path: str) -> Tuple[str, List[str]]:
+    """Return the last existing parent and a list of all intermediate directories
+    to be created for the directory passed as input.
+
+    Args:
+        path: directory to be created
+    """
+    # detect missing intermediate folders
+    intermediate_folders = []
+    last_parent = ""
+    intermediate_path = os.path.dirname(path)
+    while intermediate_path:
+        if os.path.exists(intermediate_path):
```

Comment:
This can just be `lexists`, if the deepest thing is a dangling symlink, makedirs will error anyways with ENOENT

Reply:
Done